dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/datasource/DefaultReadOnlyDataSourceManager.java

```diff
@@ -1,167 +0,0 @@
- * specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
- * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
- * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
- * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
- * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
- * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
- * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
- * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
- * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-package org.hisp.dhis.datasource;
-
-import static com.google.common.base.Preconditions.checkNotNull;
-import static org.hisp.dhis.external.conf.ConfigurationKey.CONNECTION_PASSWORD;
-import static org.hisp.dhis.external.conf.ConfigurationKey.CONNECTION_URL;
-import static org.hisp.dhis.external.conf.ConfigurationKey.CONNECTION_USERNAME;
-
-import java.beans.PropertyVetoException;
-import java.sql.SQLException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Properties;
-import javax.sql.DataSource;
-import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.lang3.StringUtils;
-import org.hisp.dhis.commons.util.DebugUtils;
-import org.hisp.dhis.external.conf.ConfigurationKey;
-import org.hisp.dhis.external.conf.DhisConfigurationProvider;
-import org.hisp.dhis.util.ObjectUtils;
-import org.springframework.beans.factory.InitializingBean;
-
-/**
- * @author Lars Helge Overland
- */
-@Slf4j
-public class DefaultReadOnlyDataSourceManager
-    implements ReadOnlyDataSourceManager, InitializingBean {
-  private static final String FORMAT_READ_PREFIX = "read%d.";
-
-  private static final String FORMAT_CONNECTION_URL = FORMAT_READ_PREFIX + CONNECTION_URL.getKey();
-
-  private static final String FORMAT_CONNECTION_USERNAME =
-      FORMAT_READ_PREFIX + CONNECTION_USERNAME.getKey();
-
-  private static final String FORMAT_CONNECTION_PASSWORD =
-      FORMAT_READ_PREFIX + CONNECTION_PASSWORD.getKey();
-
-  private static final int VAL_ACQUIRE_INCREMENT = 6;
-
-  private static final int VAL_MAX_IDLE_TIME = 21600;
-
-  private static final int MAX_READ_REPLICAS = 5;
-
-  private final DhisConfigurationProvider config;
-
-  public DefaultReadOnlyDataSourceManager(DhisConfigurationProvider config) {
-    checkNotNull(config);
-    this.config = config;
-  }
-
-  /** State holder for the resolved read only data source. */
-  private DataSource internalReadOnlyDataSource;
-
-  /** State holder for explicitly defined read only data sources. */
-  private List<DataSource> internalReadOnlyInstanceList;
-
-  @Override
-  public void afterPropertiesSet() {
-    List<DataSource> ds = getReadOnlyDataSources();
```

Comment:
This was the source of the problem, where afterPropertiesSet was not called as this class is no longer a Spring-managed bean.